exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusremotewriteexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/tidwall/wal"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+type walConfig struct {
+	Dir             string        `mapstructure:"path"`
+	CacheSize       int           `mapstructure:"cache-size"`
+	RefreshDuration time.Duration `mapstructure:"truncate-frequency"`
+}
+
+func (wc *walConfig) createWAL() (*wal.Log, error) {
```

Comment:
Nit: "create" implies it is something new, always. However, this seems to open an existing wal _or_ creates it. Consider giving the func a better name.

Reply:
That's really what it does though, it creates a new WAL every time that it is invoked, and it returns it
<img width="728" alt="Screen Shot 2021-05-05 at 12 51 28 PM" src="https://user-images.githubusercontent.com/4898263/117200723-a7694a00-ada0-11eb-8e1f-b2624e18fe83.png">